### PR TITLE
Replace deprecated `message_box_objective()` function

### PR DESCRIPTION
**Pull request by pospec4444**
Trying to address #3127 if it makes any sense there.

Replaced `message_box_objective()` with its successors in all scripts under `data/campaigns`.

### CODE REVIEW

**Comment by pospec4444** [comment]
This PR might affect TODOs here:

https://github.com/widelands/widelands/blob/07de47cde1335673d0d7902a769631a3275da038/data/scripting/messages.lua#L164-L186

Btw, RST comment on `campaign_message_with_objective()` refers to `function:: campaign_message_box()`, is that some intention?

**Comment by pospec4444** [comment]
"Road Building" images were pointing to non-existing `roadb_*.png` files, replaced with respective `road_building_*.png` links.

**Comment by frankystone** [comment]
> This PR might affect TODOs here:
> 
> https://github.com/widelands/widelands/blob/07de47cde1335673d0d7902a769631a3275da038/data/scripting/messages.lua#L164-L186

I guess we can remove the TODO.

> 
> Btw, RST comment on `campaign_message_with_objective()` refers to `function:: campaign_message_box()`, is that some intention?

Please change the RST to fit with the functions name. Then the [documentation](https://www.widelands.org/documentation/autogen_auxiliary_messages/) will show it correctly.

I think the deprecated function can also be removed?

File: data/campaigns/tutorial02_warfare.wmf/scripting/texts.lua [hunk 1]
```diff
@@ -210,10 +222,15 @@ dismantle = {
...
-   obj_name = "dismantle_sentry",
-   obj_title = _"Dismantle your north-western sentry",
-   obj_body = (
+      p(_[[You can also use this opportunity to become familiar with the other options: the heroes/rookies preference and the capacity.]])
+   )
+}
+
+obj_dismantle = {
+   name = "dismantle_sentry",
+   title = _"Soldiers, dismiss!",
```
Comment (gunchleoc) [inline data/campaigns/tutorial02_warfare.wmf/scripting/texts.lua]:
```suggestion
   title = _"Soldiers, dismissed!",
```

File: data/campaigns/tutorial02_warfare.wmf/scripting/texts.lua [hunk 2]
```diff
@@ -72,11 +72,14 @@ battlearena1 = {
...
-   h = 400,
-   obj_name = "build_battlearena",
-   obj_title = _"Build a battle arena",
-   obj_body = (
+   )
+}
+
+obj_battlearena = {
+   name = "build_battlearena",
+   title = _"Battle arena",
```
Comment (gunchleoc) [inline data/campaigns/tutorial02_warfare.wmf/scripting/texts.lua]:
```suggestion
   title = _"Battle Arena",
```

File: data/campaigns/tutorial02_warfare.wmf/scripting/texts.lua [hunk 3]
```diff
@@ -105,11 +108,14 @@ trainingcamp1 = {
...
-   h = 300,
-   obj_name = "build_trainingcamp",
-   obj_title = _"Build a training camp",
-   obj_body = (
+   )
+}
+
+obj_trainingcamp = {
+   name = "build_trainingcamp",
+   title = _"Training camp",
```
Comment (gunchleoc) [inline data/campaigns/tutorial02_warfare.wmf/scripting/texts.lua]:
```suggestion
   title = _"Training Camp",
```

File: data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua [hunk 4]
```diff
@@ -168,10 +178,14 @@ expedition3 = {
...
-   obj_title = _"Found a settlement",
-   obj_body = (
+   )
+}
+
+obj_expedition_sail = {
+   name = "found_settlement",
+   title = _"Explore",
+   number = 1,
+   body = objective_text("Found a settlement",
```
Comment (gunchleoc) [inline data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua:189]:
```suggestion
   -- TRANSLATORS: "Found" as in "founding", NOT as in "finding"
   body = objective_text("Found a settlement",
```

File: data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua [hunk 5]
```diff
@@ -200,12 +213,14 @@ expedition5 = {
...
-   obj_name = "build_iron_mine",
-   obj_title = _"Build an irone mine on the island",
-   obj_body = (
-      h1(_"Build an iron mine") ..
+   )
+}
+
+obj_expedition_iron = {
+   name = "build_iron_mine",
+   title = _"Iron mine",
```
Comment (gunchleoc) [inline data/campaigns/tutorial03_seafaring.wmf/scripting/texts.lua]:
```suggestion
   title = _"Iron Mine",
```

**Comment by gunchleoc** [comment]
> > Btw, RST comment on `campaign_message_with_objective()` refers to `function:: campaign_message_box()`, is that some intention?
> 
> Please change the RST to fit with the functions name. Then the [documentation](https://www.widelands.org/documentation/autogen_auxiliary_messages/) will show it correctly.

+1, this is a bug.

 
> I think the deprecated function can also be removed?

Yes it can, both of them.

I have also added some text suggestions.

File: data/campaigns/tutorial02_warfare.wmf/scripting/texts.lua (hunk 2 above)
Comment (pospec4444) [reply data/campaigns/tutorial02_warfare.wmf/scripting/texts.lua]: Proofreading: Tutorial texts

**Comment by pospec4444** [comment]
> > I think the deprecated function can also be removed?

> Yes it can, both of them.

Both? Which is the second one beside `message_box_objective()`? I must be missing something here...

**Comment by gunchleoc** [comment]
> > > I think the deprecated function can also be removed?
> 
> > Yes it can, both of them.
> 
> Both? Which is the second one beside `message_box_objective()`? I must be missing something here...

I thought there was a second deprecated function. Looks like I was seeing thins, so you can ignore this comment.

**Comment by gunchleoc** [comment]
Testing needed for:

- [x] Barbarians 1 (needs fix from https://github.com/widelands/widelands/pull/3793 first)
- [x] Tutorial 1
- [x] Tutorial 2
- [x] Tutorial 3

**Comment by pospec4444** [comment]
> Testing needed for Barbarians 1 (needs fix from #3793 first)

Will do.

**Comment by gunchleoc** [comment]
> > Testing needed for Barbarians 1 (needs fix from #3793 first)
> 
> Will do.

Actually, testing in Pull Requests is always done by somebody else than the person who wrote the code, to ensure a second perspective.

**Comment by pospec4444** [comment]
OK, won't do then.

(Btw, I usually test changes _before_ creating a PR.)

**Comment by gunchleoc** [comment]
> OK, won't do then.
> 
> (Btw, I usually test changes _before_ creating a PR.)

And you should too. This way 2 people will have tested it. We all acquire different habits on what we click on and things like that, so it's always good to have 2 people testing before it goes into master.

**Comment by pospec4444** [comment]
Campaign "Barbarian 01" tested, works fine.

(Speaking about testing, are there any guidelines I could follow? Nothing is listed [here](https://github.com/widelands/widelands/community).)

**Comment by gunchleoc** [comment]
> Campaign "Barbarian 01" tested, works fine.

Thanks!
 
> (Speaking about testing, are there any guidelines I could follow? Nothing is listed [here](https://github.com/widelands/widelands/community).)

I have added a link to the Widelands wiki there.

We don't have any specific protocol for manual testing. Basically, try to break it in creative ways and find bugs and if you can't, the changes are OK. It also depends a bit on what kind of change we're making.

**Pull request merged**